### PR TITLE
Add [x]Props to Tree component 

### PR DESCRIPTION
**Pull request by AdamMeza-Bentley**
## Changes

This PR continues work on #1368 by adding props to various subcomponents of `Tree.tsx` . It adds `nodeProps`, `contentProps`,  `lablelProps`,  `titleProps`,  `sublabelProps`, `checboxProps`, `iconProps`, `expanderProps` and `subTreeProps` to `TreeNode`. It also adds `iconProps` to `TreeNodeExpander`

## Testing

Added tests to `TreeNode.test.tsx` to test all the custom props.

## Docs

Added a changeset documenting the new props

### CODE REVIEW

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Tree/TreeNode.tsx:262]: The merged `className` should come after `{...checkboxProps}`, otherwise it will be overridden

**Review by Ben-Pusey-Bentley — COMMENTED** [review]
This PR should have a changeset, since this is a user facing change. Also, the PR description Docs section is empty, and should be filled.

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (AdamMeza-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx:262]: switched those in my last push

**Comment by AdamMeza-Bentley** [comment]
added changeset

**Review by Ben-Pusey-Bentley — COMMENTED** [review]
This PR should also have a name that is descriptive of the code it is changing. Something like "Adds props to `Tree` subcomponents".

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (mayank99) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx]: It's only `className` that should come after. The other props should be before. 

Specifically:

```diff
  <Box
+   as='span'
+   aria-hidden
    {...iconProps}
-   as='span'
    className={cx(
      'iui-tree-node-content-icon',
      iconProps?.className,
    )}
-   aria-hidden
  >
```

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Tree/TreeNode.tsx:1]: Some props are missing. The elements with `iui-tree-node`, `iui-tree-node-content` and `iui-sub-tree` classes need to be customizable too.

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [inline packages/itwinui-react/src/core/Tree/TreeNode.tsx:21]: I think this is supposed to be `React.ComponentProps<any>`.

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [inline packages/itwinui-react/src/core/Tree/TreeNode.tsx:97]: Think that this should also be `React.ComponentProps<any>`.

Comment (AdamMeza-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx:97]: yes that's a good catch

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (AdamMeza-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx:1]: added those

**Review by Ben-Pusey-Bentley — APPROVED** [review]
Should be good to go once the remaining comments are resolved. The failing visual test seems to be unrelated, probably a flaky test that will pass on re-run.

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Tree/TreeNode.tsx:21]: These should all be `'div'`, `'span'`, etc (matching the corresponding elements) rather than `any`

File: packages/itwinui-react/src/core/Tree/TreeNode.test.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Tree/TreeNode.test.tsx:224]: nit: maybe some of these tests could be combined, since they're all testing similar things

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx]: Fixed in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/54fc0932a2836605be25ecaba9749ecc4720c1b9).

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Tree/TreeNode.tsx:58]: More appropriate name: `subNodeProps` → `subTreeProps`

Also update the jsdoc to more clearly describe which element this affects.

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Tree/TreeNode.tsx:99]: This jsdoc should also be updated to more clearly describe which "content".

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx]: Added in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/5ef04937148b9d722e75d364ad109f8c539af188).

File: packages/itwinui-react/src/core/Tree/TreeNode.test.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.test.tsx]: Combined the tests in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/b5bb153b230aadce93b20f3d2c2bae88a68671c7).

Comment (mayank99) [reply packages/itwinui-react/src/core/Tree/TreeNode.test.tsx]: less nitty: These tests are not properly testing the className merging currently. It would be better to test both the original `iui-` class and the custom class.

See example:

https://github.com/iTwin/iTwinUI/blob/442ff49f2502aabf5f9495262a6e4f1941804d3c/packages/itwinui-react/src/core/Label/Label.test.tsx#L50-L53

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.test.tsx]: Added testing for both classes in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/0889be47f673ed465d27f49a5327a8c084c78de8).

File: packages/itwinui-react/src/core/Tree/TreeNodeExpander.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNodeExpander.tsx]: Changed in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/badf438a92a5a5cacdf862799202f40484ca905a).

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx]: Changed in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/779fc89a58412cd969726940732db0cb1ead7920).

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx]: Updated in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/437d7379210c8f1022bc0ba9e03d847f8031a3df).

Comment (mayank99) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx]: slight grammatical nit: "effects" (noun) → "affects" (verb)

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx]: Fixed in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/b684869778dbe696d7ec70e858a49e67395b4353).

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx:313]: Added in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/4d856ed86a6b73969abfe73d92c7a36dac65f411).

Comment (mayank99) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx:313]: Why is this one called `treeNodeExpanderProps` when none of the others have a `treeNode` prefix? Should just be `expanderProps`

File: .changeset/cuddly-tools-joke.md
Comment (mayank99) [inline .changeset/cuddly-tools-joke.md:5]: changeset can be updated as follows:

```suggestion
Added optional props to `TreeNode` for customization: `nodeProps`, `contentProps`, `labelProps`, `titleProps`, `sublabelProps`, `checkboxProps`, `iconProps`, `expanderProps` and `subTreeProps`.
```

also PR description can be updated.

File: packages/itwinui-react/src/core/Tree/TreeNode.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Tree/TreeNode.tsx:313]: It was named that because it was associated with the `TreeNodeExpander` component. The distinction isn't necessary though, since it is the only expander associated with the `TreeNode` component. Renamed in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/2dacdad18d90076bb2cd5c1898039a8e74eedaf7).

File: .changeset/cuddly-tools-joke.md
Comment (Ben-Pusey-Bentley) [reply .changeset/cuddly-tools-joke.md]: Committed the suggestion and updated the PR description.

**Review by mayank99 — CHANGES_REQUESTED** [review]
LGTM, but I would wait until the patch release (#2029) is completed before merging this.

Instead of approving, I'm blocking accidental merge using "request changes".

**Review by mayank99 — APPROVED** [review]

**Pull request merged**
